File: src/dataset/Java/pascals-triangle_exercism-ThomasZumsteg/gemini/GeminiFlash_PascalsTriangle_v1.java
public class PascalsTriangle {
    public static int[][] computeTriangle(int rows) {
        if (rows < 0) {
            throw new IllegalArgumentException();
        }

        int[][] triangle = new int[rows][];

        for (int i = 0; i < rows; i++) {
            triangle[i] = new int[i + 1];
            triangle[i][0] = 1;
            triangle[i][i] = 1;

            for (int j = 1; j < i; j++) {
                triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
            }
        }

        return triangle;
    }

    public static boolean isTriangle(int[][] triangle) {
        for (int i = 0; i < triangle.length; i++) {
            if (triangle[i].length != i + 1) {
                return false;
            }
            if (triangle[i][0] != 1 || triangle[i][i] != 1) {
                return false;
            }
            for (int j = 1; j < i; j++) {
                if (triangle[i][j] != triangle[i - 1][j - 1] + triangle[i - 1][j]) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: src/dataset/c/pascals-triangle_exercism-IrvinRamiA/gemini/GeminiFlash_pascals_triangle_v1.c
#include <stdlib.h>
#include "pascals_triangle.h"

void free_triangle(uint8_t **triangle, uint8_t rows) {
  if (triangle == NULL) {
    return;
  }
  for (uint8_t i = 0; i < rows; i++) {
    free(triangle[i]);
  }
  free(triangle);
}

uint8_t **create_triangle(uint8_t rows) {
  if (rows == 0) {
    return NULL;
  }

  uint8_t **triangle = (uint8_t **)malloc(rows * sizeof(uint8_t *));
  if (triangle == NULL) {
    return NULL;
  }

  for (uint8_t i = 0; i < rows; i++) {
    triangle[i] = (uint8_t *)malloc((i + 1) * sizeof(uint8_t));
    if (triangle[i] == NULL) {
      // Handle allocation failure: free previously allocated rows
      for (uint8_t j = 0; j < i; j++) {
        free(triangle[j]);
      }
      free(triangle);
      return NULL;
    }

    triangle[i][0] = 1;
    triangle[i][i] = 1;

    for (uint8_t j = 1; j < i; j++) {
      triangle[i][j] = triangle[i - 1][j - 1] + triangle[i - 1][j];
    }
  }

  return triangle;
}